asciicastv2.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+// {"version": 2, "width": 80, "height": 24, "timestamp": 1504467315, "title": "Demo", "env": {"TERM": "xterm-256color", "SHELL": "/bin/zsh"}}
+// color format: #rrggbb
+
+type ASCIICastV2Theme struct {
+	FG      string `json:"fg"`      // text color
+	BG      string `json:"bg"`      // background color
+	Palette string `json:"palette"` // list of 8 or 16 colors, separated by colon character
+}
+
+type ASCIICastV2Header struct {
+	Version       int               `json:"version"`         // (required) must be 2
+	Width         int               `json:"width"`           // (required) in columns
+	Height        int               `json:"height"`          // (required) in rows
+	Timestamp     int               `json:"timestamp"`       // (optional) unix epoch
```

Comment:
By adding `,omitempty` to the json tag you can make it so the fields are not included in the JSON if they are not set.

Reply:
ah cool :+1:  updated accordingly :)